r1cs/gadgets/crypto/src/nizk/groth16/mod.rs

```diff
@@ -190,6 +190,50 @@ for VerifyingKeyGadget<PairingE, ConstraintF, P>
         ConstraintF: Field,
         P: PairingGadget<PairingE, ConstraintF>,
 {
+
+    #[inline]
+    fn alloc_without_check<FN, T, CS: ConstraintSystem<ConstraintF>>(
+        mut cs: CS,
+        value_gen: FN,
+    ) -> Result<Self, SynthesisError>
+        where
+            FN: FnOnce() -> Result<T, SynthesisError>,
+            T: Borrow<VerifyingKey<PairingE>>,
+    {
+        value_gen().and_then(|vk| {
+            let VerifyingKey {
+                alpha_g1_beta_g2,
+                gamma_g2,
+                delta_g2,
+                gamma_abc_g1,
+            } = vk.borrow().clone();
+            let alpha_g1_beta_g2 =
+                P::GTGadget::alloc_without_check(cs.ns(|| "alpha_g1_beta_g2"), || Ok(alpha_g1_beta_g2))?;
```

Comment:
It'd worth to add a comment here, that in typical applications verification keys are provided through circuit inputs, hence no allocation checks needed.